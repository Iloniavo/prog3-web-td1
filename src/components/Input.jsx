export default function Input({setDescription, description }) {
    return (
        <input type="text" data-testid = "input"  placeholder="Add..." value={description} onChange={(e) => setDescription(e.target.value)} />
    )
};
