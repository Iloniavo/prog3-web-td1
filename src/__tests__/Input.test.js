import userEvent from "@testing-library/user-event";
import renderer from "react-test-renderer";
import Input from "../components/Input";
import {render, screen} from "@testing-library/react";

it("Render input in the screen", function(){

    const setDescriptionMock = jest.fn()
    function descriptionMock(){
        return "description"
    }
    const component = renderer.create(
        <Input setDescription={setDescriptionMock} description={descriptionMock}/>
    )

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    render(
        <Input setDescription={setDescriptionMock} description={descriptionMock}/>
    )
    
    const inputField = screen.getByPlaceholderText(descriptionMock);
    expect(inputField).toBeInTheDocument()

})