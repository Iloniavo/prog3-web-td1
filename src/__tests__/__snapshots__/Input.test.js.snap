// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check input in the screen 1`] = `
<input
  data-testid="input"
  onChange={[Function]}
  placeholder="Add..."
  type="text"
  value={[Function]}
/>
`;

exports[`Render input in the screen 1`] = `
<input
  data-testid="input"
  onChange={[Function]}
  placeholder="Add..."
  type="text"
  value={[Function]}
/>
`;
